### PR TITLE
ROC-2990: Remove validation from phone no and make it optional

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link
[ROC-2990](https://tools.hmcts.net/jira/browse/ROC-2990)

### Change description
This PR removes the mobile no validation from the phone and makes it optional. Makes it as free text up to 30 characters allowed. 

### Work checklist

- [x] Unit tests added where applicable
- [x] Task list and task completeness checks updated
- [x] Routes, page content and page flows of new features are toggled 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/app/forms/models/mobilePhone.ts [hunk 1]
```diff
@@ -29,7 +26,4 @@ export class MobilePhone implements CompletableTask {
     return this
   }
 
-  isCompleted (): boolean {
-    return !!this.number && this.number.length > 0
```
Comment (timja-kainos) [inline src/main/app/forms/models/mobilePhone.ts]: why remove this? I assume its still mandatory to provide a number?

File: src/main/features/claim/views/claimant-mobile.njk [hunk 2]
```diff
@@ -12,7 +12,7 @@
         <form method="post">
           {{ csrfProtection(csrf) }}
 
-          {{ textInput('Mobile number', 'number', form) }}
+          {{ textInput('', 'number', form) }}
```
Comment (timja-kainos) [inline src/main/features/claim/views/claimant-mobile.njk]: shouldn't there be a visually hidden label here? @radoslawn-kainos

Comment (radoslawn-kainos) [reply src/main/features/claim/views/claimant-mobile.njk]: Yup, can't leave the field with an empty label. Setting [`labelClass`](https://github.com/hmcts/cmc-common-frontend/blob/master/macros/form.njk#L1) param to `visually-hidden` should be enough. Might require modifying `textInput` macro to pass that on.

Comment (sabahirfan) [reply src/main/features/claim/views/claimant-mobile.njk]: done

**Pull request merged**